code/game/objects/structures/platforms.dm

```diff
@@ -1,156 +1,175 @@
-/*
-* Platforms
-*/
+///Slowdown applied when exiting this structure's turf
+#define ENTRY_SLOWDOWN (1<<0)
+///Slowdown applied when entering this structure's turf
+#define EXIT_SLOWDOWN (1<<1)
+
+
 /obj/structure/platform
 	name = "platform"
 	desc = "A square metal surface resting on four legs."
 	icon = 'icons/obj/structures/platforms.dmi'
 	icon_state = "platform"
-	climbable = TRUE
-	anchored = TRUE
-	density = TRUE
 	coverage = 10
-	layer = OBJ_LAYER
-	climb_delay = 20 //Leaping a barricade is universally much faster than clumsily climbing on a table or rack
-	interaction_flags = INTERACT_CHECK_INCAPACITATED //no dexterity flag so xenos can climb them
+	layer = BELOW_OBJ_LAYER
 	flags_atom = ON_BORDER
-	resistance_flags = XENO_DAMAGEABLE	//TEMP PATCH UNTIL XENO AI PATHFINDING IS BETTER, SET THIS TO INDESTRUCTIBLE ONCE IT IS - Tivi
-	obj_integrity = 1000	//Ditto
-	max_integrity = 1000	//Ditto
+	resistance_flags = RESIST_ALL
+	obj_flags = PROJ_IGNORE_DENSITY
+	///How much slowdown is applied by this platform
+	var/climb_slowdown = 0.5 SECONDS
+	///Dictates the slowdown behavior of this platform
+	var/platform_flags = ENTRY_SLOWDOWN|EXIT_SLOWDOWN
 
-/obj/structure/platform/gelida
-	coverage = 0
-	climb_delay = 5 //halved time because on gelida platforms are everywhere
-	obj_integrity = 50 //ditto
-	max_integrity = 50	//ditto
 
 /obj/structure/platform/Initialize(mapload)
 	. = ..()
-	var/image/I = image(icon, src, "platform_overlay", LADDER_LAYER, dir)//ladder layer puts us just above weeds.
-	switch(dir)
-		if(SOUTH)
-			layer = ABOVE_MOB_LAYER
-			I.pixel_y = -16
-		if(NORTH)
-			I.pixel_y = 16
-		if(EAST)
-			I.pixel_x = 16
-		if(WEST)
-			I.pixel_x = -16
-	overlays += I
-	var/static/list/connections = list(
-		COMSIG_ATOM_EXIT = PROC_REF(on_try_exit)
+	apply_overlays()
+	icon_state = ""
+
+	if(!platform_flags)
+		return
+
+	var/static/list/entry_and_exit = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_enter),
+		COMSIG_ATOM_EXITED = PROC_REF(on_exit),
 	)
-	AddElement(/datum/element/connect_loc, connections)
 
-/obj/structure/platform/proc/on_try_exit(datum/source, atom/movable/O, direction, list/knownblockers)
+	var/static/list/entry_only = list(COMSIG_ATOM_ENTERED = PROC_REF(on_enter))
+
+	var/static/list/exit_only = list(COMSIG_ATOM_EXITED = PROC_REF(on_exit))
+
+
+	AddElement(/datum/element/connect_loc, (platform_flags & ENTRY_SLOWDOWN) ? (platform_flags & EXIT_SLOWDOWN) ? entry_and_exit : entry_only : exit_only)
+
+///clears and sets overlays based on current dir
+/obj/structure/platform/proc/apply_overlays()
```

Comment:
Why not handle this through update_overlays?

Reply:
true